lessc.inc.php

```diff
@@ -1199,6 +1199,36 @@ protected function lib_percentage($arg) {
 		return array("number", $num*100, "%");
 	}
 
+	// Mix color with white in variable proportion. It is the same as calling mix(#ffffff, @color, @weight)
+	// tint(@color, [@weight: 50%]);
+	// Returns: color
+	// http://lesscss.org/functions/#color-operations-tint
+	protected function lib_tint($args) {
+		$white = ['color', 255, 255, 255];
+		if ($args[0] == 'color') {
+			return $this->lib_mix([ 'list', ',', [$white, $args] ]);
+		} elseif ($args[0] == "list" && count($args[2]) == 2) {
+			return $this->lib_mix([ $args[0], $args[1], [$white, $args[2][0], $args[2][1]] ]);
+		} else {
+			$this->throwError("tint expects (color, weight)");
+		}
+	}
+
+	// Mix color with black in variable proportion. It is the same as calling mix(#000000, @color, @weight)
+	// shade(@color, [@weight: 50%]);
+	// Returns: color
+	// http://lesscss.org/functions/#color-operations-shade
+	protected function lib_shade($args) {
+		$black = ['color', 0, 0, 0];
+		if ($args[0] == 'color') {
+			return $this->lib_mix([ 'list', ',', [$black, $args] ]);
+		} elseif ($args[0] == "list" && count($args[2]) == 2) {
+			return $this->lib_mix([ $args[0], $args[1], [$black, $args[2][0], $args[2][1]] ]);
+		} else {
+			$this->throwError("tint expects (color, weight)");
```

Comment:
"shade expects (color, weight)"

would probably be a better message ;)

Reply:
Thanks, I've fixed this as part of the merge.